annotation/tests/test_tasks_services.py

```diff
@@ -0,0 +1,311 @@
+from unittest.mock import Mock, call, patch
+
+import pytest
+from fastapi import HTTPException
+from sqlalchemy.orm import Session
+
+from annotation.errors import FieldConstraintError
+from annotation.jobs.services import ValidationSchema
+from annotation.models import File, ManualAnnotationTask
+from annotation.tasks.services import (
+    check_cross_annotating_pages,
+    validate_files_info,
+    validate_task_info,
+    validate_user_actions,
+    validate_users_info,
+)
+
+TEST_PARAMS_INVALID_FILES = (
+    (
+        {
+            "file_id": 1,
+            "job_id": 2,
+            "pages": [1, 2, 4],
+        },
+        Mock(spec=File, pages_number=3),
+        "pages ({4}) do not belong to file",
+    ),
+    (
+        {
+            "file_id": 1,
+            "job_id": 2,
+            "pages": [1, 2, 3],
+        },
+        None,
+        "file with id 1 is not assigned for job 2",
+    ),
+)
+
+
+TEST_PARAMS_VALIDATE_USER_ACTIONS = (
+    (
+        True,
+        {1},
+        set(),
+        set(),
+        False,
+        True,
+        "Missing `annotation_user_for_failed_pages` param "
+        "for failed pages.",
+    ),
+    (
+        True,
+        set(),
+        {2},
+        set(),
+        True,
+        False,
+        "Missing `validation_user_for_reannotated_pages` param "
+        "for edited pages.",
+    ),
+    (
+        True,
+        set(),
+        set(),
+        set(),
+        True,
+        True,
+        "Validator did not mark any pages as failed, thus "
+        "`annotation_user_for_failed_pages` param should be null.",
+    ),
+    (
+        True,
+        {1},
+        set(),
+        set(),
+        True,
+        True,
+        "Validator did not edit any pages, thus "
+        "`validation_user_for_reannotated_pages` param should be null.",
+    ),
+    (
+        True,
+        {1},
+        {2},
+        {3},
+        True,
+        True,
+        "Cannot finish validation task. There are not processed pages: " "{3}",
+    ),
+)
+
+
+@pytest.fixture(scope="function")
+def db_session():
+    return Mock(spec=Session)
+
+
+@pytest.fixture(scope="function")
+def task_info():
+    return {
+        "is_validation": True,
+        "user_id": 1,
+        "job_id": 2,
+    }
+
+
+@pytest.mark.parametrize(
+    ("validation_type", "is_validation"),
+    (
+        (ValidationSchema.validation_only, True),
+        ("other_type", False),
+        ("other_type", True),
+    ),
+)
+def test_validate_task_info_positive_case(
```

Comment:
No need for 'positive_case'. If there is no description, it is considered a success test case.